src/Containers/vectorized_product_iterator.jl

```diff
@@ -0,0 +1,32 @@
+"""
+    struct VectorizedProductIterator{T}
+        prod::Iterators.ProductIterator{T}
+    end
+
+Same as `Base.Iterators.ProductIterator` except that it is independent
+on the `IteratorSize` of the elements of `prod.iterators`.
```

Comment:
Could this description be made more self contained? Currently it refers to an undocumented internal field of `ProductIterator`.